utils/firestore.js

```diff
@@ -2098,6 +2113,16 @@ const getNotificationSpecById = async (id) => {
     return snapshot.empty ? null : snapshot.docs[0].data();
 }
 
+const getNotificationSpecByCategoryAndAttempt = async (category = "", attempt = "") => {
```

Comment:
do we have notifications where the category or attempt would be null? we would be better off null checking them and saving an attempt at searching the database if the parameters are null?

Reply:
No, we don't have notifications with `null` as category or attempt. So I didn't put those checks.
I agree that we cannot assume the function is correctly used/called, so thorough checking is needed.